src/js/tracks/text-track-settings.js

```diff
@@ -273,25 +242,184 @@ class TextTrackSettings extends Component {
   }
 
   /**
+   * Create a <select> element with configured options.
+   *
+   * @private
+   * @return {Element}
+   * @method createElSelect_
+   */
+  createElSelect_(key) {
+    const config = selectConfigs[key];
+    const id = config.id.replace('%s', this.id_);
+
+    return [
+      createEl(
+        'label',
+        {className: 'vjs-label', textContent: config.label},
+        {for: id}),
+      createEl(
+        'select',
+        {id},
+        undefined,
+        config.options.map(o => {
+          return createEl('option', {textContent: o[1], value: o[0]});
+        }))
+    ];
+  }
+
+  /**
+   * Create color elements for the component
+   *
+   * @private
+   * @return {Element}
+   * @method createElColors_
+   */
+  createElColors_() {
+    return createEl(
+      'div',
+      {className: 'vjs-tracksettings-colors'},
+      undefined,
+      [
+        createEl(
```

Comment:
maybe create all of these elements outside of the main `createEl` call so that this code is more understandable (as they will all have to have names, and you can add comments)